src/hls-adapter.ts

```diff
@@ -791,6 +792,12 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     this._onTrackChanged(textTrack);
   }
 
+  private _onSubtitleFragProcessed(): void {
+    this._hls.subtitleTrack = -1;
+    this._firstHideTextTrack = false;
+    this._hls.off(Hlsjs.Events.SUBTITLE_FRAG_PROCESSED, this._onSubtitleFragProcessed, this._hls);
```

Comment:
are there any other cases where the event listener becomes irrelevant and should be removed ? like when the adapter is reset ?

Reply:
add it under remove bindings that called in reset function